spot_rl_experiments/spot_rl/envs/skill_manager.py

```diff
@@ -0,0 +1,233 @@
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import os
+import time
+from typing import Tuple
+
+import numpy as np
+import rospy
+from spot_rl.envs.gaze_env import GazeController, construct_config_for_gaze
+from spot_rl.envs.nav_env import WaypointController, construct_config_for_nav
+from spot_rl.envs.place_env import PlaceController, construct_config_for_place
+from spot_rl.utils.geometry_utils import (
+    is_pose_within_bounds,
+    is_position_within_bounds,
+)
+from spot_rl.utils.utils import (
+    get_waypoint_yaml,
+    nav_target_from_waypoint,
+    place_target_from_waypoint,
+)
+from spot_wrapper.spot import Spot
+
+DOCK_ID = int(os.environ.get("SPOT_DOCK_ID", 520))
+
+
+# SpotSkillManager? SpotSkillExecutor? SpotSkillController?
+class SpotSkillManager:
+    def __init__(self):
+        # We will probably receive a config as part of the constructor
+        # And eventually that specifies the set of skills that we can instantiate and execute
+        self.__init_spot()
+        self.__initiate_controllers()
+
+        # Power on the robot
+        # TODO: Check if this can be moved outside of env
```

Comment:
Hi, Kavit, do you want to clean up this?

Reply:
Thank you @jimmytyyang for taking care of this.